lib/matplotlib/cbook.py

```diff
@@ -2661,3 +2661,170 @@ def __exit__(self, exc_type, exc_value, traceback):
                     os.rmdir(path)
                 except OSError:
                     pass
+
+
+class _FuncInfo(object):
+    """
+    Class used to store a function
+
+    Each object has:
+    * The direct function (direct)
+    * The inverse function (inverse)
+    * A boolean indicating whether the function
+      is bounded in the interval 0-1 (bounded_0_1)
+
+    """
+    def __init__(self, direct, inverse, bounded_0_1):
+        self.direct = direct
+        self.inverse = inverse
+        self.bounded_0_1 = bounded_0_1
+
+    def copy(self):
+        return _FuncInfo(self.direct,
+                         self.inverse,
+                         self.bounded_0_1)
+
+
+class _StringFuncParser(object):
+    """
+    A class used to convert predefined strings into
+    _FuncInfo objects, or to directly obtain _FuncInfo
+    properties.
+
+    """
+
+    _funcs = {}
+    _funcs['linear'] = _FuncInfo(lambda x: x,
+                                 lambda x: x,
+                                 True)
+    _funcs['quadratic'] = _FuncInfo(lambda x: x**2,
+                                    lambda x: x**(1. / 2),
+                                    True)
+    _funcs['cubic'] = _FuncInfo(lambda x: x**3,
+                                lambda x: x**(1. / 3),
+                                True)
+    _funcs['sqrt'] = _FuncInfo(lambda x: x**(1. / 2),
+                               lambda x: x**2,
+                               True)
+    _funcs['cbrt'] = _FuncInfo(lambda x: x**(1. / 3),
```

Comment:
Hmm, wondering now if there's a way to generically support any function if the name is passed in using a lookup of numpy math function names...(the implementation of the functions would also be pitched to numpy, which maybe should be happening anyway instead of lambdas...)